app/components/edit/EditAddress.jsx

```diff
@@ -51,45 +65,54 @@ class EditAddress extends Component {
           state_province: '',
         };
       }
+      console.log(`no address: ${newAddr}`);
       return { noLocation: !state.noLocation, address: newAddr };
     }, () => {
-      const { address } = this.state;
+      const { index, address } = this.state;
+      console.log(`INDEX ${index}`);
       updateAddress(address);
     });
   }
 
   render() {
-    const { address = {} } = this.props;
+    const { index, item: address } = this.props;
+    const htmlID = `address${index}`;
     const { noLocation } = this.state;
     return (
-      <AddressForm
-        handleAddressChange={this.handleAddressChange}
-        handleNoLocationChange={this.handleNoLocationChange}
-        noLocation={noLocation}
-        address={address}
-      />
+      <div>
+        <label>
+          <input
+            type="checkbox"
+            className="input-checkbox"
+            checked={noLocation}
+            onChange={this.handleNoLocationChange}
+          />
+            No Physical Location
+        </label>
+
+        <AddressForm
+          handleAddressChange={this.handleAddressChange}
+          noLocation={noLocation}
+          address={address}
+          htmlID={htmlID}
+        />
+      </div>
     );
   }
 }
 
 const AddressForm = ({
-  noLocation, handleNoLocationChange, handleAddressChange, address,
+  noLocation, handleAddressChange, address, htmlID,
 }) => (
   <li key="address" className="edit--section--list--item">
-    <label htmlFor="address">Address</label>
-    <label>
-      <input
-        type="checkbox"
-        className="input-checkbox"
-        checked={noLocation}
-        onChange={handleNoLocationChange}
-      />
-        No Physical Location
-    </label>
+
+
     {!noLocation
         && (
           <div>
+            <label htmlFor={htmlID}>Address</label>
             <input
+              id={htmlID}
```

Comment:
BTW, this is not a good idea because in HTML, every `id` needs to be unique, and if you put the same ID on multiple `<input>` elements, there's no way to distinguish between them.